build_test.go

```diff
@@ -549,6 +549,152 @@ version = "1.1.1"
 		}))
 	})
 
+	it("ignore working-directory setting and writes launch.toml (API<0.8)", func() {
+		builder.On("Build", mock.Anything).Return(libcnb.BuildResult{
+			BOM: &libcnb.BOM{Entries: []libcnb.BOMEntry{
+				{
+					Name:     "test-launch-bom-entry",
+					Metadata: map[string]interface{}{"test-key": "test-value"},
+					Launch:   true,
+				},
+				{
+					Name:     "test-build-bom-entry",
+					Metadata: map[string]interface{}{"test-key": "test-value"},
+				},
+			}},
+			Labels: []libcnb.Label{
+				{
+					Key:   "test-key",
+					Value: "test-value",
+				},
+			},
+			Processes: []libcnb.Process{
+				{
+					Type:             "test-type",
+					Command:          "test-command-in-dir",
+					Default:          true,
+					WorkingDirectory: "/my/directory/",
+				},
+			},
+			Slices: []libcnb.Slice{
+				{
+					Paths: []string{"test-path"},
+				},
+			},
+		}, nil)
+
+		libcnb.Build(builder,
+			libcnb.WithBOMLabel(true),
+			libcnb.WithArguments([]string{commandPath, layersPath, platformPath, buildpackPlanPath}),
+			libcnb.WithTOMLWriter(tomlWriter),
+		)
+
+		Expect(tomlWriter.Calls[0].Arguments[0]).To(Equal(filepath.Join(layersPath, "launch.toml")))
+		Expect(tomlWriter.Calls[0].Arguments[1]).To(Equal(libcnb.LaunchTOML{
+			Labels: []libcnb.Label{
+				{
+					Key:   "test-key",
+					Value: "test-value",
+				},
+			},
+			Processes: []libcnb.Process{
+				{
+					Type:    "test-type",
+					Command: "test-command-in-dir",
+					Default: true,
+				},
+			},
+			Slices: []libcnb.Slice{
+				{
+					Paths: []string{"test-path"},
+				},
+			},
+			BOM: []libcnb.BOMEntry{
+				{
+					Name:     "test-launch-bom-entry",
+					Metadata: map[string]interface{}{"test-key": "test-value"},
+					Launch:   true,
+				},
+			},
+		}))
+	})
+
+	it("writes launch.toml with working-directory setting(API>=0.8)", func() {
+		var b bytes.Buffer
+		err := buildpackTOML.Execute(&b, map[string]string{"APIVersion": "0.8"})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(ioutil.WriteFile(filepath.Join(buildpackPath, "buildpack.toml"), b.Bytes(), 0600)).To(Succeed())
+		builder.On("Build", mock.Anything).Return(libcnb.BuildResult{
+			BOM: &libcnb.BOM{Entries: []libcnb.BOMEntry{
+				{
+					Name:     "test-launch-bom-entry",
+					Metadata: map[string]interface{}{"test-key": "test-value"},
+					Launch:   true,
+				},
+				{
+					Name:     "test-build-bom-entry",
+					Metadata: map[string]interface{}{"test-key": "test-value"},
+				},
+			}},
+			Labels: []libcnb.Label{
+				{
+					Key:   "test-key",
+					Value: "test-value",
+				},
+			},
+			Processes: []libcnb.Process{
+				{
+					Type:             "test-type",
+					Command:          "test-command-in-dir",
+					Default:          true,
+					WorkingDirectory: "/my/directory/",
+				},
+			},
+			Slices: []libcnb.Slice{
+				{
+					Paths: []string{"test-path"},
+				},
+			},
+		}, nil)
```

Comment:
```suggestion
			Processes: []libcnb.Process{
				{
					Type:             "test-type",
					Command:          "test-command-in-dir",
					Default:          true,
					WorkingDirectory: "/my/directory/",
				},
			},
		}, nil)
```

Reply:
Could we make appropriate changes in the rest of the places to reduce the testing code to be limited to the process working dir bits?